docs/public/configuration/gui/building-custom-gui.md

```diff
@@ -0,0 +1,170 @@
+# Build a custom Zonemaster-GUI installation package
+
+## Background
+
+If you follow the [Zonemaster-GUI installation instructions] you can install an
+official package, and skip the instructions in this document.
+
+However, if you have created a custom [config.ts], done theme settings in
+[tsconfig.json] or done some [theme updates] then you must create a custom
+installation package for your custom installation. That can be acheived by
+following the steps below.
+
+It is also important to state that even though the installation package is
+created on Ubuntu 22.04 below, the resulting installation package can be
+installed on at least all OSs supported in the
+[Zonemaster-GUI installation instructions].
+
+## Prepare build environment
+
+Start by creating a build environment. Here we assume and base it on
+[Ubuntu] version 22.04. The instructions will probably work with other versions
+of Ubuntu, or with other Linux distributions or other OSs, but then you might
+need to adapt some of the commands. Note however that it is important that the
+system fully supports [npm].
+
+### Install toolchain
+
+1. Make a clean installation of Ubuntu 22.04.
+
+2. Update the package database and install Curl and Git.
+
+```sh
+sudo apt-get update
+sudo apt-get install curl git
+```
```

Comment:
The presentation is better if code blocks in lists are indented to the same depth as the list items.
```suggestion
   ```sh
   sudo apt-get update
   sudo apt-get install curl git
   ```
```